### ISSUE TITLE
Thanks for your interest.

### ISSUE THREAD

**Issue by szgy66** [minshu-kim/Neural-Image-Stitching#9]
Thanks for your interest.
The warped image can be obtained by stitching a blank image (consisting of 0s) and a warped image.

For example, in our [stitch.py](https://github.com/minshu-kim/Neural-Image-Stitching/blob/main/stitch.py) code, 1) modify this [line](https://github.com/minshu-kim/Neural-Image-Stitching/blob/77a96f582ab96435c796e74419a89dba955acfd6/stitch.py#L76) as shown below to obtain the aligned images:
```
        fea_ref_w *= ref_mask.cpu()
        fea_tgt_w *= tgt_mask.cpu()
        fea = torch.cat([torch.zeros_like(fea_tgt_w), fea_tgt_w], dim=1).cuda()
```

and for masking valid regions, 2) revise this [line](https://github.com/minshu-kim/Neural-Image-Stitching/blob/77a96f582ab96435c796e74419a89dba955acfd6/stitch.py#L207) as below:
```
    pred = ((pred + 1)/2).clamp(0,1) * stit_mask.reshape(b, 1, *sizes) * tgt_mask
```

_Originally posted by @minshu-kim in https://github.com/minshu-kim/Neural-Image-Stitching/issues/7#issuecomment-2345393090_

**Comment by szgy66** [comment]
> ```
>           Thanks for your interest.
> ```
> 
> The warped image can be obtained by stitching a blank image (consisting of 0s) and a warped image.
> 
> For example, in our [stitch.py](https://github.com/minshu-kim/Neural-Image-Stitching/blob/main/stitch.py) code, 1) modify this [line](https://github.com/minshu-kim/Neural-Image-Stitching/blob/77a96f582ab96435c796e74419a89dba955acfd6/stitch.py#L76) as shown below to obtain the aligned images:
> 
> ```
>         fea_ref_w *= ref_mask.cpu()
>         fea_tgt_w *= tgt_mask.cpu()
>         fea = torch.cat([torch.zeros_like(fea_tgt_w), fea_tgt_w], dim=1).cuda()
> ```
> 
> and for masking valid regions, 2) revise this [line](https://github.com/minshu-kim/Neural-Image-Stitching/blob/77a96f582ab96435c796e74419a89dba955acfd6/stitch.py#L207) as below:
> 
> ```
>     pred = ((pred + 1)/2).clamp(0,1) * stit_mask.reshape(b, 1, *sizes) * tgt_mask
> ```
> 
> _Originally posted by @minshu-kim in [#7 (comment)](https://github.com/minshu-kim/Neural-Image-Stitching/issues/7#issuecomment-2345393090)_

Thanks for your reply! I got the warpped image by your answer. But I got the warpped image with size [932, 1604], when the ref and tgt images with size [756, 1008]. How can I get an image of the same size？

**Comment by minshu-kim** [comment]
The stitched image has a different size from the reference and target images as the image alignment requires a large-sized canvas (or image template).

Nevertheless, if you want to resize the stitched image, you can crop or interpolate it to the size of a reference image. I recommend cropping ROI from the stitched image as interpolation of an image causes blurred and aliased image quality.
